.github/workflows/intel-oneapi.yml

```diff
@@ -0,0 +1,167 @@
+name: linux-intel-oneapi
+on: [push, pull_request]
+
+defaults:
+  run:
+    shell: bash -leo pipefail {0}
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    env:
+      CC: icc
+      FC: ifort
+      CXX: icpc
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+
+    steps:
+
+    - name: install-intel
+      run: |
+        cd /tmp
+        wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+        sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+        rm GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+        echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
+        sudo apt update
+        sudo apt-get install intel-oneapi-dev-utilities intel-oneapi-mpi-devel intel-oneapi-openmp intel-oneapi-compiler-fortran intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic
+        echo "source /opt/intel/oneapi/setvars.sh" >> ~/.bash_profile
+        sudo apt-get install doxygen
+
+    - name: cache-netcdf
+      id: cache-netcdf
+      uses: actions/cache@v2
+      with:
+        path: ~/netcdf
+        key: netcdf-c-${{ runner.os }}-4.7.4-intel
+
+    - name: build-hdf5
+      if: steps.cache-netcdf.outputs.cache-hit != 'true'
+      run: |
+        export CC=mpiicc
```

Comment:
I like to use "set -x" at the top of all my build commands, but I see you have removed them. I wonder if there's a way we could turn that one in one place for all commands...

Reply:
I added it. You can invoke the shell with custom commands. I had to do this with Intel to load the environment with a login shell (`-l`) so that it sourced `.bash_profile`. Here I just added `-x`.